src/aioswitcher/api/__init__.py

```diff
@@ -162,20 +172,32 @@ async def _login(self) -> Tuple[str, SwitcherLoginResponse]:
 
         """
         timestamp = current_timestamp_to_hexadecimal()
-        if (
-            self._device_type
-            and self._device_type == DeviceType.BREEZE
+        if bool(self._token):
+            packet = packets.LOGIN_TOKEN_PACKET_TYPE2.format(
+                self._token, timestamp, self._device_id
+            )
+        elif (
+            self._device_type == DeviceType.BREEZE
             or self._device_type == DeviceType.RUNNER
             or self._device_type == DeviceType.RUNNER_MINI
         ):
-            packet = packets.LOGIN2_PACKET_TYPE2.format(timestamp, self._device_id)
+            packet = packets.LOGIN_PACKET_TYPE2.format(timestamp, self._device_id)
```

Comment:
```suggestion
            packet = packets.LOGIN2_PACKET_TYPE2.format(timestamp, self._device_id)
```
It seems like you change the login for existing devices, are you sure this change is correct and not a typo?

Reply:
I did not change the login for existing devices. I renamed it to be more clear as requested by @TomerFi.
If you look over the value in this line you will see it evaluates to the same value.
So we have those:

\# first login packet for Type1 devices
LOGIN_PACKET_TYPE1

\# first login packet for Type2 devices
LOGIN_PACKET_TYPE2

\# first login packet for token-based Type2 devices
LOGIN_TOKEN_PACKET_TYPE2

\# second login packet for token-based Type2 devices
LOGIN2_TOKEN_PACKET_TYPE2

LOGIN2_PACKET_TYPE2 is renamed to LOGIN_PACKET_TYPE2.

This is how it looks now:
```
        if bool(self._token):
            packet = packets.LOGIN_TOKEN_PACKET_TYPE2.format(
                self._token, timestamp, self._device_id
            )
        elif (
            self._device_type == DeviceType.BREEZE
            or self._device_type == DeviceType.RUNNER
            or self._device_type == DeviceType.RUNNER_MINI
        ):
            packet = packets.LOGIN_PACKET_TYPE2.format(timestamp, self._device_id)
        else:
            packet = packets.LOGIN_PACKET_TYPE1.format(timestamp, self._device_key)
        signed_packet = sign_packet_with_crc_key(packet)

        logger.debug("sending a login packet")
        self._writer.write(unhexlify(signed_packet))
        response = await self._reader.read(1024)

        if bool(self._token):
            packet = packets.LOGIN2_TOKEN_PACKET_TYPE2.format(
                self._device_id, timestamp, self._token
            )
            signed_packet = sign_packet_with_crc_key(packet)
            logger.debug("sending a login2 packet")
            self._writer.write(unhexlify(signed_packet))
            response = await self._reader.read(1024)
```

So the original code is ok.